src/streaming-client.ts

```diff
@@ -94,12 +98,19 @@ export class RevAiStreamingClient extends EventEmitter {
         return new BufferedDuplex(this.requests, this.responses, { readableObjectMode: true });
     }
 
+    /**
+     * Signals to the api that you have finished sending data.
+     */
+    public finish(): void {
```

Comment:
I would call this `end()` and rename the current end to `unsafeEnd()` and document that `unsafeEnd` immediately closes the connection and that no output will be returned.

Reply:
since `end` is the contract that our (non-existent) customers would be using to close their connections

Reply:
I like this suggestion. It avoids a breaking change and makes sense as an implementation